src-docs/spec/sendRealtimeData.md

```diff
@@ -0,0 +1,48 @@
+# sendRealtimeData()
+
+```js
+window.webxdc.sendRealtimeData(data);
+```
+
+**Note that this API is experimental and not fully settled (April 2024)**
+
+Send `Uint8Array` data to other realtime peers for this app. 
+You must first call [`setRealtimeListener`](./setRealtimeListener.md) 
```

Comment:
good point -- i had thought about it but one iteration before the one you commented, we had `setRealtimeListener` return a promise that resolved only when a first peer is connected.  So there was a way to do a `sendRealtimeData` after waiting for the promise to resolve and be optimistic that at least one peer will see it. 
Now, that we removed the promise, we could as well trigger the setup of the realtime machinery on `sendRealtimeData` as well even though it's almost guaranteed the sent data will not land anywhere if we are just starting the machinery.  @link2xt what do you think?  You were the most critical about the promise IIRC.

Reply:
also, in a prior incarnation `setRealtimeListener` the method was called `joinRealtimeSession` so maybe something like that name wouldn't feel so weird then?

Reply:
after some more thought, i suggest we just trigger the "join realtime" machinery on both methods, so there is no ordering dependency between the methods anymore.  I guess if one starts an app and joins realtime connections, then leaves the app, then starts it again, the connections might still be there and `sendRealtimeData` succeeds without calling `setRealtimeListener` first.

Reply:
if we keep connection setup totally implicit i wonder how would an app be able to ever force disconnection from peers?  Ensuring that no code in an app still tries to send data may not be easy which would implicitely set it up again.  I think it's better to have some explicit way to manage connectivity. If we also want to avoid having a dependency of top-level send and receive APIs then we should consider this API: 

`const realtimeSession = webxdc.joinRealtimeSession(receiveCallback)`
and then having `realtimeSession.sendData()` and `realtimeSession.disconnect()` methods. 
There would be no way to send data or leave a session without first joining, and it's natural that after `disconnect` the session is invalidated and you need to start fresh to reconnect and send/receive data. 
This API design gives both explicit control over connectivity, and does not need a reminder to only send data after you registered a receive-listener.

Reply:
Here is my last comment implemented in a separate PR: https://github.com/webxdc/website/pull/78/files